app/src/main/java/com/example/android/codelabs/paging/ui/SearchRepositoriesViewModel.kt

```diff
@@ -42,20 +39,45 @@ class SearchRepositoriesViewModel(private val repository: GithubRepository) : Vi
     @Volatile
     private var lastQueryValue: String? = null
     @Volatile
-    private var lastSearchResult: Flow<PagingData<Repo>>? = null
+    private var lastSearchResult: Flow<PagingData<UiModel>>? = null
 
     /**
      * Search a repository based on a query string.
      */
-    fun searchRepo(queryString: String): Flow<PagingData<Repo>> {
+    fun searchRepo(queryString: String): Flow<PagingData<UiModel>> {
         val result = lastSearchResult
-        if(queryString == lastQueryValue && result != null){
+        if (queryString == lastQueryValue && result != null) {
             return result
         }
         lastQueryValue = queryString
-        val newResult = repository.getSearchResultStream(queryString)
+        val newResult: Flow<PagingData<UiModel>> = repository.getSearchResultStream(queryString)
+                .map { pagingData -> pagingData.map { UiModel.RepoItem(it) as UiModel } }
+                .map {
+                    it.insertSeparators { before, after ->
```

Comment:
without the above map, before/after will be RepoItems, so you don't need to do all the checks

you may end up needing to be specific about generics, but I think that's clearer than adding all the uses of 'is' as 'as'